protocols/gossipsub/src/behaviour.rs

```diff
@@ -586,20 +586,29 @@ where
         Ok(true)
     }
 
-    /// Publishes a message with multiple topics to the network.
+    /// Publishes a message to a topic in the network.
     pub fn publish<H: Hasher>(
         &mut self,
         topic: Topic<H>,
```

Comment:
```suggestion
        topic: impl Into<TopicHash>,
```

Would it make sense to generalise the API to this instead of duplicating the function?
This would allow you to pass either a topic or a topic hash.

Reply:
cc @AgeManning What do you think of this API change?

Reply:
I like this approach, as it extends the existing API without overcomplicating it. Changed this PR to match this comment and added a changelog item.